src/order/filter/at_top_bot.lean

```diff
@@ -516,4 +516,18 @@ lemma tendsto_at_top_of_monotone_of_subseq {ι ι' α : Type*} [preorder ι] [pr
   tendsto u at_top at_top :=
 tendsto_at_top_of_monotone_of_filter h (map_ne_bot hl) (tendsto_map' H)
 
+lemma tendsto_neg_at_top_at_bot {α : Type*} [nonempty α] [decidable_linear_ordered_add_comm_group α] :
```

Comment:
You shouldn't need `nonempty` for this one, in the same way that it is not needed for the next one.

Reply:
Indeed ! To clarify, all this class mess is because I initially proved most lemmas only about real numbers, and then translated them, but that gives way too strong requirements